bin/push-docker

```diff
@@ -12,5 +11,19 @@ echo $COMMIT
 echo $REPO
 echo $TAG
 echo '--'
+echo "Trying Docker login..."
 docker login -u $DOCKER_USER -p $DOCKER_PASS
-docker push $REPO
\ No newline at end of file
+if [ ! $? -eq 0 ]; then
+    exit
+fi
+
+status=1
+count=1
+# Retry up to 5 times total if failed Docker push
+until [ ${status} -eq 0 ] || [ $count -gt 5 ]; do
+    echo "Trying Docker push..."
+    docker push $REPO
+    status=$?
+    count=$((count+1))
+    sleep 15
```

Comment:
This is fine for this script, but the pattern in general is flawed, because after the last failed retry it will needlessly delay for 15 seconds and then exit with success code.

Reply:
In fact it even sleeps after a successful try, which is unnecessary.